File: platforms/swift/examples/core/pass/add_model.swift
import FragmentColor

let mesh = Mesh()
try mesh.addVertex(
    try Vertex([0.0, 0.5, 0.0]).set(Vertex.nORMAL, [0.0, 0.0, 1.0]).set(Vertex.uV0, [0.5, 1.0]),
)
try mesh.addVertex(
    try Vertex([-0.5, -0.5, 0.0]).set(Vertex.nORMAL, [0.0, 0.0, 1.0]).set(Vertex.uV0, [0.0, 0.0]),
)
try mesh.addVertex(
    try Vertex([0.5, -0.5, 0.0]).set(Vertex.nORMAL, [0.0, 0.0, 1.0]).set(Vertex.uV0, [1.0, 0.0]),
)

let template = Material.pbr()?.baseColor([0.85, 0.4, 0.2, 1.0])
let pass = Pass("scene")

let m1 = Model(mesh.clone(), template.clone())
m1.translate([-1.0, 0.0, 0.0])
pass.addModel(m1)

let m2 = Model(mesh, template)
m2.translate([1.0, 0.0, 0.0])
pass.addModel(m2)

File: platforms/kotlin/examples/core/pass/add_model.kt
import org.fragmentcolor.*

val renderer = Renderer()
val mesh = Mesh()
mesh.addVertex( Vertex.new(listOf(0.0f, 0.5f, 0.0f)).set(Vertex.NORMAL, floatArrayOf(0.0f, 0.0f, 1.0f)).set(Vertex.UV0, listOf(0.5f, 1.0f)), )
mesh.addVertex( Vertex.new(listOf(-0.5f, -0.5f, 0.0f)).set(Vertex.NORMAL, floatArrayOf(0.0f, 0.0f, 1.0f)).set(Vertex.UV0, listOf(0.0f, 0.0f)), )
mesh.addVertex( Vertex.new(listOf(0.5f, -0.5f, 0.0f)).set(Vertex.NORMAL, floatArrayOf(0.0f, 0.0f, 1.0f)).set(Vertex.UV0, listOf(1.0f, 0.0f)), )

val template = Material.pbr(renderer).baseColor(listOf(0.85f, 0.4f, 0.2f, 1.0f))
val pass = Pass("scene")

val m1 = Model(mesh.clone(), template.clone())
m1.translate(listOf(-1.0f, 0.0f, 0.0f))
pass.addModel(m1)

val m2 = Model(mesh, template)
m2.translate(listOf(1.0f, 0.0f, 0.0f))
pass.addModel(m2)